### ISSUE TITLE
Inverted horizontal scrolling on OS X

### ISSUE THREAD

**Issue by nbolton** [symless/synergy-core#617]
**Imported issue:**
- Author: eglas...@hotmail.com (Google user)
- Date: 2010-07-24 00:00:00
- Legacy ID: 630

> THIS IS NOT FOR SUPPORT, PLEASE USE  
> "http://groups.google.com/group/synergy-plus":http://groups.google.com/group/synergy-plus  
> NO NEED TO SUBSCRIBE TO THE GROUP TO RECEIVE ANSWERS.  
> 
> IMPORTANT: Please research your problem thoroughly before opening  
> a new ticket. The following tips are provided to help you.  
> 
> Tips:  
> - Please ensure that all labels below are correct.  
> - It may be useful to attach debug output ("http://tinyurl.com/y9w4t5h":http://tinyurl.com/y9w4t5h )  
> - Have you searched for an existing or similar issue?  
> - If you have a question, please post on the mailing list.  
> - Have you selected the correct issue template?  
> 
> Note: You can delete the above text block before posting.  
> 
> What list of steps will reproduce the problem?  

Install & run synergy on two mac os x 10.6  

> What version of operating system, and Synergy+ are you using?  

server - Mac OS X 10.6.4  
client - Mac OS X 10.6.3  

> What is the expected behaviour, and what happens instead?  

The horizontal scrolling appears kinda 'inverted': when scrolling to the left on the server machine, picture or anything else scrolls to the right on the client and vice versa.  

Server machine is a MacBook with multitouch touchpad which supports scrolling in any direction.  

> Is there a way of temporarily working around your problem?  

I've not found it  

> Would you like to solve the problem yourself and submit a patch?  

I do not know how

**Comment by the-wes** [comment]
Marking this as an Enhancement because I believe this should be an option. I don't think this should be forced one way or the other.

**Comment by ruudwelten** [comment]
I think this issue should be marked as a bug since it still exists and it has nothing to do with Mac OS X's 'natural' scrolling option. If the 'natural' option is toggled the vertical scrolling direction changes on the client as well (as expected) but the horizontal direction switches as well, so again: it scrolls in the opposite direction as it does on the server.

**Server**
iMac 27" 3.5GHz i7 / 8GB RAM / 500GB SSD (Late 2013)
Mac OS X 10.9.3
Synergy Pro 1.7.2 (without SSL encryption because that does not work)
Apple Magic Mouse / Apple Mighty Mouse / Generic Mouse

**Client**
MacBook Pro 15" Unibody 2.53GHz Core 2 Duo / 8GB RAM / 2x 500GB SSD (Late 2008)
Mac OS X 10.8.5
Synergy Pro 1.7.2
Apple Mighty Mouse / Built-in Apple Trackpad / Generic Mouse

**Comment by ktorn** [comment]
I'm experiencing this too. Since the horizontal scrolling doesn't behave like the vertical scrolling, at least from a UX perspective I agree with @ruudwelten in calling it a bug.

**Comment by tolgraven** [comment]
This is definitely 100% a bug. No reason to make it an option, control is had through the 'natural scrolling' option.

**Comment by jnoarm** [comment]
I too am experiencing this, is there any update on potential fix ?

**Comment by the-wes** [comment]
Caving in to the pressure and changing the status to "bug" :)

**Comment by AlexKinney** [comment]
This has just started happening to me as well. Not sure what caused it. Yesterday it was working, today it's not :/

reintstalling synergy on both machines didn't fix it

Anyone find a fix?

**Comment by davecotter** [comment]
bug.  no need for option.  just please put a negative sign in there and make a new build?

**Comment by jmason86** [comment]
Ditto on the problem. iMac Late 2015 (server) to rMBP Late 2013 (client) using magic mouse. Synergy version 1.8.2-stable-36cd521 built 2016 September 02.

**Comment by zepheiryan** [comment]
Facing the same problem.  I've been building Synergy on OS X from source locally; on the v1.8.2-stable branch, I made these changes and distributed to clients - this seems to have fixed it for now, but like others, it wasn't previously a constant problem for me, so grains of salt and all.  I'll update if it later fails.

@davecotter - more like remove some negative signs ;)

```diff
diff --git a/src/lib/platform/OSXScreen.cpp b/src/lib/platform/OSXScreen.cpp
index a216dc1..b85b3ab 100644
--- a/src/lib/platform/OSXScreen.cpp
+++ b/src/lib/platform/OSXScreen.cpp
@@ -671,7 +671,7 @@ OSXScreen::fakeMouseWheel(SInt32 xDelta, SInt32 yDelta) const
                CGEventRef scrollEvent = CGEventCreateScrollWheelEvent(
                        NULL, kCGScrollEventUnitLine, 2,
                        mapScrollWheelFromSynergy(yDelta),
-                       -mapScrollWheelFromSynergy(xDelta));
+                       mapScrollWheelFromSynergy(xDelta));
                
         // Fix for sticky keys
         CGEventFlags modifiers = m_keyState->getModifierStateAsOSXFlags();
@@ -1022,7 +1022,7 @@ OSXScreen::handleSystemEvent(const Event& event, void*)
                        }
 
                        if (xScroll != 0 || yScroll != 0) {
-                               onMouseWheel(-mapScrollWheelToSynergy(xScroll),
+                               onMouseWheel(mapScrollWheelToSynergy(xScroll),
                                                                mapScrollWheelToSynergy(yScroll));
                        }
                }
```

**Comment by ben-ng** [comment]
I'm still running into this issue, and it's pretty annoying.

**Comment by AWTrost** [comment]
Also having this issue... and as its been bugged for 2.5 years its kind of frustrating that its not fixed (and hasn't been updated for 6 months)...  sorry for being that guy #sorrynotsorry

**Comment by jmason86** [comment]
Installed 1.8.4 and still have the issue. Is the fix for this more complicated that just inserting a minus sign in the right place?

**Comment by AlexKinney** [comment]
+1 hoping this can get fixed

**Comment by jdebardi** [comment]
Still an issue in 1.8.7-stable. Server/Client both running macOS Sierra 10.12.3.
Appears to always follow the inverse of the 'natural scroll' setting on the server. Client 'natural scroll' setting has no effect.

**Comment by yeison** [comment]
+1

**Comment by g8keepa** [comment]
+1

**Comment by ruudwelten** [comment]
2010-07-24 is the first date in this thread... It's been redefined as a bug over two years ago! How is this still not fixed? I've bought multiple versions but am still having this issue.

**Comment by davecotter** [comment]
it's not fixed because this app is abandoned by the developer

**Comment by samrae7** [comment]
I don't understand why this bug has been closed. I purchased this product two days ago and still has this issue ( version 1.11.1)

**Comment by davecotter** [comment]
@nbolton please re-open this bug, it has been here for ages, has gotten dozens of complaints. just because no dev ever took it up does NOT mean users don't want it fixed!

**Comment by PeterKeffer** [comment]
Same issue here!

**Comment by Fabrice6767** [comment]
Same issue also here :-) OMG this was first mentioned 7 years ago ?

**Comment by brunorsini** [comment]
Same issue here, so annoying 😔

**Comment by andreacndn** [comment]
The fix for this problem is in progress, it will be included in 1.14.1
SYNERGY-988

**Comment by RevenantX** [comment]
Seems it fixed in 1.14.1

**Comment by dneprqa** [comment]
Fixed in 1.14.1
